flowpipe/graph.py

```diff
@@ -107,30 +107,33 @@ def add_node(self, node):
             LogObserver.push_message(
                 "Node '{0}' is already part of this Graph".format(node.name))
 
-    def evaluate(self, threaded=False, submission_delay=0.1):
+    def evaluate(self, threaded=False, submission_delay=0.1, raise_after=3):
```

Comment:
I'd keep raise_after=None because 3 seems an arbitrary choice and we don't really have enough data to know if that is actually a good value (probably depends on the specs, the os etc). Or is there a deeper reason for the 3?

Reply:
There is no deeper reason for the value 3. I think the default behavior here comes down to a matter of taste. Since I know little about threading and/or multiprocessing, I tend to prefer the cautious route, throwing the error. Nevertheless, this is unfounded as I don't know how to provoke such a situation.

Go with whatever default makes more sense to you. I'm fine with both. In the end, the option is there, so if someone prefers one behavior over the other, they can get what they prefer.